components/ping_ai_copilot/extension/content/content_scripts/rephraser.js

```diff
@@ -0,0 +1,440 @@
+        buttonContainer.classList.remove('dark-shadow')
+        buttonContainer.classList.add('light-shadow')
+    }
+    else {
+        buttonContainer.classList.remove('light-shadow');
+        buttonContainer.classList.add('dark-shadow');
+    }
+    const img = document.createElement('img');
+    img.src = chrome.runtime.getURL(`extension/assets/rephrase-${iconColor}.svg`);
+    img.alt = 'Rephrase';
+    img.style.width = '17px';
+    img.style.height = '17px';
+    img.style.objectFit = 'contain';
+
+    buttonContainer.appendChild(img);
+
+    const showPill = () => {
+        if (hasRephrasedBefore) {
+            buttonContainer.style.display = 'none';
+            createPillContainer(textBox, img);
+        }
+    };
+
+    const hidePill = () => {
+        if (pillContainer) {
+            pillContainer.remove();
+            pillContainer = null;
+        }
+        buttonContainer.style.display = 'flex';
+        buttonContainer.style.justifyContent = 'center';
+        buttonContainer.style.alignItems = 'center';
+    };
+
+    buttonContainer.addEventListener('mouseenter', () => {
+        if (!isFetching) {
+            if (hasRephrasedBefore) {
+                showPill();
+            } else {
+                img.src = chrome.runtime.getURL(`extension/assets/rephrase-${iconColor}-hover.svg`);
+            }
+        }
+    });
+
+    buttonContainer.addEventListener('mouseleave', (event) => {
+        if (!isFetching) {
+            if (hasRephrasedBefore) {
+                // Check if the mouse is moving to the pill
+                const pillRect = pillContainer ? pillContainer.getBoundingClientRect() : null;
+                if (!pillRect || 
+                    event.clientX < pillRect.left || 
+                    event.clientX > pillRect.right || 
+                    event.clientY < pillRect.top || 
+                    event.clientY > pillRect.bottom) {
+                    hidePill();
+                }
+            } else {
+                img.src = chrome.runtime.getURL(`extension/assets/rephrase-${iconColor}.svg`);
+            }
+        }
+    });
+
+    const containerHeight = img.style.height;
+    const containerWidth = img.style.width;
+
+    buttonContainer.style.left = `${rect.right - parseInt(containerWidth) - 32 + window.scrollX}px`;
+    buttonContainer.style.top = `${rect.bottom - parseInt(containerHeight) - 20 + window.scrollY}px`;
+
+    document.body.appendChild(buttonContainer);
+
+    rephraseButton = buttonContainer;
+
+    buttonContainer.addEventListener('click', async (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        if (isFetching) {
+            // If fetching, stop the process and revert to original text
+            undoRephraseText(textBox, img);
+        } else {
+            // Start rephrasing process
+            rephraseText(textBox, img);
+        }
+    });
+
+    // Observe changes to the DOM to detect when the text box is removed or hidden
```

Comment:
also check here if the text box is resized.
 In case of chatgpt I saw that the rephraser icons does not always come in the corner. it might be because of resizing of text box or some other issue
<img width="806" alt="Screenshot 2024-09-17 at 12 10 15 PM" src="https://github.com/user-attachments/assets/96fba1d0-273b-4653-b85e-60c1b2d74cd5">

Reply:
![Screenshot_20240920_045942](https://github.com/user-attachments/assets/d0a4be3d-48c9-44f1-ba90-546431345c81)
![Screenshot_20240920_050041](https://github.com/user-attachments/assets/a2faa9ba-6487-47a9-aeb8-b5130ed634f3)
![Screenshot_20240920_050128](https://github.com/user-attachments/assets/ccae3707-fa15-4b50-91a2-6c58948c6540)